lib/irc/BridgedClient.js

```diff
@@ -666,8 +666,8 @@ BridgedClient.prototype._joinChannel = function(channel, key, attemptCount) {
     if (Object.keys(this.unsafeClient.chans).indexOf(channel) !== -1) {
         return Promise.resolve(new IrcRoom(this.server, channel));
     }
-    if (channel.indexOf("#") !== 0) {
-        // PM room
+    if (!/[#!&+]/.test(channel)) {
```

Comment:
This looks insufficent compared to the last test. `/[#&+]/` is checking if the channel has these characters anywhere in it's name rather than at the start, whereas the previous line checked the first character.

![image](https://user-images.githubusercontent.com/2072976/62616355-2d740a80-b907-11e9-8758-374bf2a4ca4b.png)

Reply:
Good point, I've fixed that.